mrtt-ui/src/components/RestorationAimsForm/RestorationAimsCheckboxGroupWithLabel.js

```diff
@@ -0,0 +1,92 @@
+import { Controller } from 'react-hook-form'
+import { FormLabel } from '@mui/material'
+import { FormQuestionDiv } from '../../styles/forms'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import CheckboxGroup from '../CheckboxGroup/CheckboxGroup'
+import getOptionsValuesAndLabels from '../../library/getOptionsValuesAndLabels'
+import StakeholderBenefitsInputs, { stakeholdersPropType } from './StakeholderBenefitsInputs'
+
+const RestorationAimsCheckboxGroupWithLabel = ({
+  fieldName,
+  options,
+  question,
+  reactHookFormInstance,
+  stakeholders
+}) => {
+  const {
+    control: formControl,
+    getValues: getFormValues,
+    setValue: setFormValue
+  } = reactHookFormInstance
+  const optionsValueLabels = getOptionsValuesAndLabels(options)
+  const id = `${fieldName}-label`
+
+  const handleAimChangeAndResetStakeholdersIfAppropriate = ({
+    event,
+    isOtherChecked,
+    otherValue,
+    selectedValues
+  }) => {
+    const isAimChecked = event.target.checked
+    const aimStakeholderBenefitsInForm = getFormValues(fieldName)?.aimStakeholderBenefits
+    const aimStakeholderBenefitsReset = {
+      ...aimStakeholderBenefitsInForm,
+      [event.target.value]: undefined
+    }
+
+    setFormValue(fieldName, {
+      selectedValues,
+      otherValue,
+      isOtherChecked,
+      aimStakeholderBenefits: isAimChecked
+        ? aimStakeholderBenefitsInForm
+        : aimStakeholderBenefitsReset
+    })
+  }
+
```

Comment:
hard to tell the data format from this, but if you look in postman, its be best I could come up with the data analyst in mind given the time constraints